packages/kit/src/runtime/client/router.js

```diff
@@ -109,12 +109,13 @@ export class Router {
 		const handle_mousemove = (event) => {
 			clearTimeout(mousemove_timeout);
 			mousemove_timeout = setTimeout(() => {
-				trigger_prefetch(event);
+				event.target.dispatchEvent(new CustomEvent('trigger_prefetch', { bubbles: true }));
```

Comment:
We could add a comment above here explaining the reason of emitting a custom event.

Reply:
I'm afraid of being verbose, but would somethink like the following work?

```js
// the custom event guarantees that event.composedPath() doesn't return an empty array
```

Or, in a positive form

```js
// the custom event guarantees that event.composedPath() returns an array of EventTarget objects
```

Reply:
Looking at the other comments a more active tone might be preferable

```js
// ensure event.composedPath() returns an array of EventTarget objects
```

Reply:
Yeah the active tone comment sounds nice. But I wouldn't worry too much about that, feel free to explain as clearly as needed, as long as it's not one giant essay. (That's a lot of as)